runtime/onert/api/include/nnfw_experimental_train.h

```diff
@@ -0,0 +1,182 @@
+ * 2. nnfw_train
+ *
+ * If you want to inference with the same session, you can use the following API
+ * - nnfw_run
+ *
+ */
+
+//////////////////////////////////////////////
+// Essential APIs for training
+//////////////////////////////////////////////
+
+/**
+ * @brief Training information to prepare training
+ */
+typedef struct nnfw_train_inf
+{
+  /** Learning rate */
+  float learning_rate = 0.001;
+  /** Batch size */
+  uint32_t batch_size = 1;
+} nnfw_train_inf;
+
+NNFW_STATUS nnfw_create_train_session(nnfw_train_session **session);
+
+/**
+ * @brief Prepare session to be ready for training
+ *
+ * @note  If we can get training information from model, info may be deprecated.
+ *        Or we will change type to pointer and overwrite training information if not null.
+ *        (Policy is not decided yet)
+ *
+ * @param[in] session The session to be prepared for training
+ * @param[in] info    Training information.
+ *                    Default value is {learning_rate: 0.001, batch_size: 1}
+ * @return  @c NNFW_STATUS_NO_ERROR if successful
+ */
+NNFW_STATUS nnfw_prepare_train(nnfw_train_session *session, nnfw_train_inf info);
+
+/**
+ * @brief Train the model
+ *
+ * @param[in]   session         The session to train the model
+ * @param[in]   inputs          The input buffers for training
+ * @param[in]   input_infos     The shape and type of input buffers
+ * @param[in]   expecteds       The expected buffers for training
+ * @param[in]   expected_infos  The shape and type of expected buffers
+ * @param[out]  loss            The loss value after training
+ * @return @c NNFW_STATUS_NO_ERROR if successful
+ */
+NNFW_STATUS nnfw_train(nnfw_train_session *session, void **inputs, nnfw_tensorinfo *input_infos,
+                       void **expecteds, nnfw_tensorinfo *expected_infos, float *loss);
+
+/**
+ * @brief Inference with the trained model
+ *
+ * @param[in] session       The session to inference with the trained model
+ * @param[in] inputs        The input buffers for inference
+ * @param[in] input_infos   The shape and type of input buffers
+ * @param[in] outputs       The output buffers for inference
+ * @param[in] output_infos  The shape and type of output buffers
+ * @return @c NNFW_STATUS_NO_ERROR if successful
+ */
+NNFW_STATUS nnfw_inference(nnfw_train_session *session, void **inputs, nnfw_tensorinfo *input_infos,
+                           void **outputs, nnfw_tensorinfo *output_infos);
+
+//////////////////////////////////////////////
+// Optional APIs for training
+//////////////////////////////////////////////
+
+/**
+ * @brief Get the number of training model inputs
+ *        This function should be called after {@link nnfw_prepare_train}
+ *
+ * @param[in]   session The session to get the number of training model inputs
+ * @param[out]  count   The number of training model inputs
+ * @return @c NNFW_STATUS_NO_ERROR if successful
+ */
+NNFW_STATUS nnfw_get_input_count(nnfw_train_session *session, uint32_t *count);
```

Comment:
@hseok-oh We may need use different name to distinguish the same name for inference (`nnfw_session` series).